public/js/all-breaches/all-breaches.js

```diff
@@ -85,112 +86,113 @@ function makeBreaches(breachArray, breachCardWrapper) {
+    makeBreachInfoSpans("breach-key", LocalizedBreachCardStrings.CompromisedAccounts, wrapper);
     makeBreachInfoSpans("breach-value", breach.PwnCount, wrapper);
 
     // compromised data
-    makeBreachInfoSpans("breach-key", breach.String.CompromisedData, wrapper);
+    makeBreachInfoSpans("breach-key", LocalizedBreachCardStrings.CompromisedData, wrapper);
     makeBreachInfoSpans("breach-value", breach.DataClasses, wrapper);
 
     // add link at bottom of card
     wrapper = makeDiv("breach-card-link-wrap", breachInfoWrapper);
-    makeBreachInfoSpans("more-about-this-breach", breach.String.MoreInfoLink, wrapper);
+    makeBreachInfoSpans("more-about-this-breach", LocalizedBreachCardStrings.MoreInfoLink, wrapper);
   }
 
   breachCardWrapper.appendChild(fragment);
   breachCardWrapper.classList.toggle("hide-breaches");
-  const breachLogos = new breachImages();
   breachLogos.lazyLoad();
-  document.addEventListener("scroll", breachLogos.lazyLoad);
-  window.addEventListener("resize", breachLogos.lazyLoad);
-  window.addEventListener("orientationchange", breachLogos.lazyLoad);
-  FOUCHotFix();
-  return breachArray;
-}
+  const loader = document.getElementById("breaches-loader");
 
-function FOUCHotFix() {
-  document.onreadystatechange = () => {
-    if (document.readyState === "complete") {
-      const loader = document.getElementById("breaches-loader");
-      loader.classList = ["hide"];
-    }
-  };
+  loader.classList = ["hide"];
+  return breaches;
 }
 
-(async() => {
-
-  if (document.getElementById("all-breaches")) {
-    const breachCardWrapper = document.getElementById("all-breaches");
+function initBreaches() {
+  const breachCardWrapper = document.querySelector("#all-breaches");
+  if (breachCardWrapper) {
     const breachWrapper = document.getElementById("breach-array-json");
+    const {LocalizedBreachCardStrings, breaches} = JSON.parse(breachWrapper.dataset.breachArray);
+
+    const breachLogos = new breachImages();
+    document.addEventListener("scroll", breachLogos.lazyLoad);
+    window.addEventListener("resize", breachLogos.lazyLoad);
+    window.addEventListener("orientationchange", breachLogos.lazyLoad);
 
-    const breachArray = JSON.parse(breachWrapper.dataset.breachArray);
+    const doBreaches = (arr) => {
+      makeBreaches(arr, LocalizedBreachCardStrings, breachCardWrapper, breachLogos);
+    };
+
+    const firstFifteen = breaches.slice(0,15);
+    doBreaches(firstFifteen);
 
-    makeBreaches(breachArray, breachCardWrapper);
-    const showAllBreaches = document.getElementById("show-all-breaches");
     const noResultsBlurb = document.getElementById("no-results-blurb");
 
     const fuzzyFindInput = document.getElementById("fuzzy-find-input");
     const fuzzyFinder = document.getElementById("fuzzy-form");
 
-    showAllBreaches.addEventListener("click", (e) => {
+    const [fuzzyShowAll, showHiddenBreaches] = document.querySelectorAll(".show-all-breaches");
```

Comment:
Are we guaranteed that these will always come back in this order? Or should we do it more like this?

```js
for (const showAllBreachesElement of document.querySelectorAll(".show-all-breaches")) {
  if (showAllBreachesElement.classList.contains("fuzzy-find-input")) {
    // do the code for fuzzyShowAll and return
  }
  // do the code for showHiddenBreaches (and return)
}
```

Reply:
We could do this but then we'd still need to assign them to variables because they are also used down the line.

Reply:
Hmm .... maybe a helper function then? But not sure how much of the vars we'd have to pass in to make it work? Might not be worth it.

```js
 function addShowAllListeners() {
   const showAllElements = {};
   for (const showAllBreachesElement of document.querySelectorAll(".show-allbreaches")) {
     if (showAllBreachesElement.classList.contains("fuzzy-find-input")) {
       showAllBreachesElement.addEventListener("click", (e) => {
         e.preventDefault();
      fuzzyFindInput.value = "";
      doBreaches(breaches);
      noResultsBlurb.classList = [""];
      fuzzyShowAll.classList = ["fuzzy-find-show-breaches"];
     });
     showAllElements.fuzzyShowAll = showAllBreachesElement;
   } else {
     showAllBreachesElement.addEventListener("click", (e) => {
    doBreaches(breaches);
    showHiddenBreaches.classList.add("hide");
     });
     showAllElements.showHiddenBreaches = showAllBreachesElement;
   }
 }
 return showAllElements;
 }

const { fuzzyShowAll, showHiddenBreaches } = addShowAllListeners();
```